alica_engine/include/engine/scheduler/Scheduler.h

```diff
@@ -0,0 +1,34 @@
+#pragma once
+
+#include "engine/scheduler/Job.h"
+#include "engine/scheduler/JobQueue.h"
+#include <vector>
+#include <condition_variable>
+#include <mutex>
+#include <thread>
+#include <atomic>
+
+namespace alica
+{
+    class AlicaEngine;
+
+namespace scheduler
```

Comment:
From what I can tell your scheduler solution is much more complicated then it needs to be:

- Jobs do not need to be handled by shared_ptr, they could be handled by values
- RunningPlans do not need to retain a weak_ptr to their initJobs or terminationJobs, those are only used in the context of generating init/terminate jobs.
- Jobs do not need prerequisites. If your traverse the plan-tree correctly, you can generate one single vector of jobs that has exactly the right order of jobs. The jobs generated in that way should be added as a batch to the scheduler to avoid hitting that mutex too often. The only reason to then keep prerequisite is due to multi-threading the job queue. See the comment there.
- Ideally you could prove that your implementation is correct with respect to the order with a test.